src/engine.rs

```diff
@@ -142,9 +142,12 @@ where
         let len = log_batch.finish_populate(self.cfg.batch_compression_threshold.0 as usize)?;
         let block_handle = {
             let mut writer = Writer::new(log_batch, sync, start);
+            // Snapshot and clear the current perf context temporarily, so the write group
+            // leader will collect the perf context diff later.
+            let mut perf_context = get_perf_context().with(|pc| pc.take());
             if let Some(mut group) = self.write_barrier.enter(&mut writer) {
                 let now = Instant::now();
-                let _t = StopWatch::new_with(&ENGINE_WRITE_LEADER_DURATION_HISTOGRAM, now);
+                let _t = StopWatch::new_with(&*ENGINE_WRITE_LEADER_DURATION_HISTOGRAM, now);
                 for writer in group.iter_mut() {
                     ENGINE_WRITE_PREPROCESS_DURATION_HISTOGRAM.observe(
```

Comment:
write_leader_wait_duration should be recorded here. The leader wait duration doesn't equal followers' wait duration.

Reply:
But I don't think it makes much sense to distinguish leader and followers' wait duration. The follower will not finish until a whole write group is done. And because of this, the `log_write_duration` of each writer is actually the total duration of all writers of the write group.

If we need to be more accurate and record each followers' wait duration, we also have to record separated `write_duration` for each writer, and also record how long it waits until other writers finish writing before syncing. This seems over-complicated to me...

Reply:
I agree with you that only the group's write time needs to be captured. So the end time to calculate wait_duration should be the same. But the start time is different. More specifically the follower wait time is longer than the leader wait time, in the worst case the difference will equal last write group's write time.

It's kind of hard to get a local perf event in your current implementation. I suggest changing the old `start_time` field in Writer to `end_time`, and recording the wait duration individually after exiting group.

Reply:
Got it. Let me try to fix it.